### PR TITLE
[Token] Fix two fields in the token transfer event

### PR DESCRIPTION
**Pull request by areshand**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4494)

### CODE REVIEW

**Review by movekevin — APPROVED** [review]

**Review by chloeqjz — APPROVED** [review]

File: aptos-move/framework/aptos-token/sources/token_transfers.move [hunk 1]
```diff
@@ -102,7 +102,7 @@ module aptos_token::token_transfers {
         event::emit_event<TokenOfferEvent>(
             &mut borrow_global_mut<PendingClaims>(sender_addr).offer_events,
             TokenOfferEvent {
-                to_address: sender_addr,
+                to_address: receiver,
```
Comment (bowenyang007) [inline aptos-move/framework/aptos-token/sources/token_transfers.move:105]: can we use token_offer_id.to_addr?

File: aptos-move/framework/aptos-token/sources/token_transfers.move [hunk 2]
```diff
@@ -102,7 +102,7 @@ module aptos_token::token_transfers {
...
                 token_id,
```
Comment (bowenyang007) [inline aptos-move/framework/aptos-token/sources/token_transfers.move:106]: can we use token_offer_id.token_id?

File: aptos-move/framework/aptos-token/sources/token_transfers.move [hunk 3]
```diff
@@ -173,7 +173,7 @@ module aptos_token::token_transfers {
         event::emit_event<TokenCancelOfferEvent>(
             &mut borrow_global_mut<PendingClaims>(sender_addr).cancel_offer_events,
             TokenCancelOfferEvent {
-                to_address: sender_addr,
+                to_address: receiver,
                 token_id,
```
Comment (bowenyang007) [inline aptos-move/framework/aptos-token/sources/token_transfers.move:177]: Same as above.

File: aptos-move/aptos-transactional-test-harness/tests/aptos_test_harness/table.exp [hunk 4]
```diff
@@ -70,39 +51,13 @@ key 0x3::token::Collections {
...
-    data:    "f75daa73fc071f93593335eb9033da804777eb94491650dd3f095ce6f778acb60b6170746f735f70756e6b730a6765656b5f746f6b656e00000000000000000100000000000000"
-}
-{
-    key:     0d00000000000000f75daa73fc071f93593335eb9033da804777eb94491650dd3f095ce6f778acb6
-    seq_num: 0
-    type:    0x3::token_transfers::TokenClaimEvent
-    data:    "9c3b634ac05d0af393e0f93b9b19b61e7cac1c519f566276aa0c6fd15dac12aaf75daa73fc071f93593335eb9033da804777eb94491650dd3f095ce6f778acb60b6170746f735f70756e6b730a6765656b5f746f6b656e00000000000000000100000000000000"
-}mutable inputs after call: local#0: 0
-return values: 0
+Error: Failed to execute transaction. ExecutionStatus: status MISSING_DATA of type Execution
```
Comment (davidiw) [inline aptos-move/aptos-transactional-test-harness/tests/aptos_test_harness/table.exp:60]: Why did we land this PR?

**Pull request merged**